pyomo/contrib/solver/tests/unit/test_base.py

```diff
@@ -136,48 +87,43 @@ def test_class_method_list(self):
-        self.assertEqual(self.instance.add_block(None), None)
-        self.assertEqual(self.instance.remove_variables(None), None)
-        self.assertEqual(self.instance.remove_parameters(None), None)
-        self.assertEqual(self.instance.remove_constraints(None), None)
-        self.assertEqual(self.instance.remove_block(None), None)
-        self.assertEqual(self.instance.set_objective(None), None)
-        self.assertEqual(self.instance.update_variables(None), None)
-        self.assertEqual(self.instance.update_parameters(), None)
-
-        with self.assertRaises(NotImplementedError):
-            self.instance._get_primals()
-
-        with self.assertRaises(NotImplementedError):
-            self.instance._get_duals()
-
         with self.assertRaises(NotImplementedError):
-            self.instance._get_reduced_costs()
-
-    @unittest.mock.patch.multiple(base.PersistentSolverBase, __abstractmethods__=set())
-    def test_context_manager(self):
-        with base.PersistentSolverBase() as self.instance:
-            self.assertTrue(self.instance.is_persistent())
             self.assertEqual(self.instance.set_instance(None), None)
+        with self.assertRaises(NotImplementedError):
             self.assertEqual(self.instance.add_variables(None), None)
+        with self.assertRaises(NotImplementedError):
             self.assertEqual(self.instance.add_parameters(None), None)
+        with self.assertRaises(NotImplementedError):
             self.assertEqual(self.instance.add_constraints(None), None)
+        with self.assertRaises(NotImplementedError):
             self.assertEqual(self.instance.add_block(None), None)
+        with self.assertRaises(NotImplementedError):
             self.assertEqual(self.instance.remove_variables(None), None)
+        with self.assertRaises(NotImplementedError):
             self.assertEqual(self.instance.remove_parameters(None), None)
+        with self.assertRaises(NotImplementedError):
             self.assertEqual(self.instance.remove_constraints(None), None)
+        with self.assertRaises(NotImplementedError):
             self.assertEqual(self.instance.remove_block(None), None)
+        with self.assertRaises(NotImplementedError):
             self.assertEqual(self.instance.set_objective(None), None)
+        with self.assertRaises(NotImplementedError):
             self.assertEqual(self.instance.update_variables(None), None)
+        with self.assertRaises(NotImplementedError):
             self.assertEqual(self.instance.update_parameters(), None)
+        with self.assertRaises(NotImplementedError):
+            self.instance._get_primals()
+        with self.assertRaises(NotImplementedError):
+            self.instance._get_duals()
+        with self.assertRaises(NotImplementedError):
+            self.instance._get_reduced_costs()
+
+    def test_context_manager(self):
+        with base.PersistentSolverBase() as self.instance:
```

Comment:
why `self.instance` and not just `instance`?

Reply:
For not a single good reason. I'll fix it.